compiler/rustc_resolve/src/lib.rs

```diff
@@ -1499,6 +1499,9 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             self.tcx.sess.time("check_reexport_ambiguities", || {
                 self.check_reexport_ambiguities(exported_ambiguities)
```

Comment:
```suggestion
                self.check_hidden_glob_reexports(exported_ambiguities)
```
Both the old check and the new check can be performed in a single pass.